midend/interpreter.cpp

```diff
@@ -911,7 +903,7 @@ void ExpressionEvaluator::postorder(const IR::MethodCallExpression* expression)
 
                 if (expression->arguments->size() == 1) {
                     // 1-argument extract method
-                    if (!fixed || !argType->is<IR::Type_Header>()) {
+                    if (!fixed && !argType->is<IR::Type_Header>()) {
```

Comment:
why make this change?

Reply:
In the case when it is installed || some states are not included in the program, in particular the state associated with varbit

Reply:
I do not understand this explanation, and I believe that this change is also wrong.

Reply:
I fixed it in a new commit